Feat/cleanup autogenerated permissions

Mishra-coder opened the pull request:
### Description

This PR adds automatic cleanup of outdated autogenerated permission files.  
Previously, old `.toml` files stayed in the `autogenerated` folder even after commands were removed, causing stale permissions to load at runtime.

### What’s changed

- Compare existing autogenerated files with the current command list
- Remove files that no longer match any command

### Why

Prevents stale permission files, reduces confusion, and keeps generated output accurate.

### Related Issue
Fixes  https://github.com/tauri-apps/tauri/issues/14320

Mishra-coder commented:
@FabianLars,@Legend-Master  PR is ready for mearge please review and suggest changes if required..

Legend-Master commented on crates/tauri-utils/src/acl/build.rs:
```diff
@@ -316,6 +316,25 @@ commands.deny = ["{command}"]
 
+  use std::collections::HashSet;
+
+  let expected_files: HashSet<String> =
```
We can push a list of generated files from this time near the `let out_path = path.join(format!("{command}.toml"));` above

Legend-Master commented on crates/tauri-utils/src/acl/build.rs:
```diff
@@ -316,6 +316,25 @@ commands.deny = ["{command}"]
+      if let Ok(file_type) = entry.file_type() {
+        if file_type.is_file() {
+          let file_name = entry.file_name().to_string_lossy().to_string();
+          if !expected_files.contains(&file_name) {
```
I think we probably should be doing path check here

Legend-Master reviewed (commented):
Thanks! Also do you mind adding a change file?

> https://github.com/tauri-apps/tauri/blob/dev/.changes/README.md

Mishra-coder commented:
@Legend-Master Please review and let me know if anything else is needed.

Legend-Master commented on package.json:
```diff
@@ -25,6 +25,7 @@
     "example:api:dev": "pnpm run --filter \"api\" tauri dev"
   },
   "devDependencies": {
+    "@changesets/cli": "^2.29.7",
```
We have this setup in the CI (also we use convector not changesets), please remove this

Legend-Master commented on crates/tauri-utils/src/acl/build.rs:
```diff
@@ -286,12 +286,15 @@ pub fn autogenerate_command_permissions(
     let toml = format!(
       r###"{license_header}# Automatically generated - DO NOT EDIT!
 {schema_entry}
+
```
Please revert this

Legend-Master commented on .changes/cleanup-autogenerated-permissions.md:
```diff
@@ -0,0 +1,6 @@
+---
+'tauri-utils': 'patch'
+'tauri-cli': 'patch'
```
```suggestion
'tauri-utils': 'patch:enhance'
'tauri': 'patch:enhance'
'tauri-build': 'patch:enhance'
'tauri-plugin': 'patch:enhance'
```

Legend-Master commented on crates/tauri-utils/src/acl/build.rs:
```diff
@@ -316,6 +321,24 @@ commands.deny = ["{command}"]
 
+  use std::collections::HashSet;
+
+  let expected_files: HashSet<PathBuf> = generated_files.into_iter().collect();
```
Let's just use the set above, replacing `let mut generated_files = Vec::new();`

Legend-Master reviewed (commented):
Please revert unrelated changes and address the review comments above

Legend-Master commented on crates/tauri-utils/src/acl/build.rs:
```diff
@@ -316,6 +322,20 @@ commands.deny = ["{command}"]
       .push(format!("deny-{slugified_command}"));
   }
 
+  if let Ok(entries) = fs::read_dir(path) {
```
```suggestion
  // Remove outdated permission files
  if let Ok(entries) = fs::read_dir(path) {
```

Legend-Master reviewed (dismissed):
Thanks!

And to merge this, we'll need all your commits to be signed

> https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

Legend-Master commented on crates/tauri-utils/src/acl/build.rs:
```diff
@@ -281,11 +281,15 @@ pub fn autogenerate_command_permissions(
     "".to_string()
   };
 
+  use std::collections::HashSet;
```
Let's import this in the top of this file

Legend-Master commented:
You'll also need to sign all your past commits

> https://stackoverflow.com/questions/41882919/is-there-a-way-to-gpg-sign-all-previous-commits

FabianLars commented:
Seems like they are signed but you didn't upload the key into github yet https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-gpg-key-to-your-github-account

idk what's with the first commit as that one is associated with a different account (but i assume that's still you judging by the similarity to your first name). if you can't fix that i can bypass the merge requirements as an exception if Tony signs off on the changes.

Mishra-coder committed:
feat(acl): remove outdated autogenerated permission files

This commit implements automatic cleanup of outdated autogenerated
permission files when commands are removed from the build configuration.

Changes:
- Added HashSet to track generated permission files
- Implemented cleanup logic to remove files not in the current command list
- Maintains write_if_changed optimization benefits

Fixes #14320

Mishra-coder committed:
Merge branch 'dev' into feat/cleanup-autogenerated-permissions
